cmd/podman/create.go

```diff
@@ -413,9 +443,12 @@ func parseCreateOpts(ctx context.Context, c *cli.Context, runtime *libpod.Runtim
 	}
 
 	// EXPOSED PORTS
-	portBindings, err := cc.ExposedPorts(c.StringSlice("expose"), c.StringSlice("publish"), c.Bool("publish-all"), data.ContainerConfig.ExposedPorts)
-	if err != nil {
-		return nil, err
+	var portBindings map[nat.Port][]nat.PortBinding
+	if data != nil {
```

Comment:
At some point we should consolidate all the image based changes into a single function and call it at the beginning, to avoid all of these `if data != nil` blocks

Reply:
Does not necessarily need to happen in this PR, as this entire section of code really needs to be refactored - we have extensive duplication in three places (create, run, varlink create) that needs to be folded into here